### PR TITLE
rss-ecomm-4_10

### PR DESCRIPTION
**Pull request by valeryiay**
In progress

1. Task: [RSS-ECOMM-4_10: Fetch and Display Cart Items Using commercetools API](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint4/RSS-ECOMM-4_10.md)
2. Screenshot:
Screenshot 2024-06-18 163603

3. Change description

- [x]  Make use of the commercetools API to fetch the list of items present in the user's shopping cart.
- [x] Present this list of items on the user interface, making sure that the name, image, and individual price, as well as the total cost for all quantities of each item, are included.

4. Change rationale
To provide a detailed summary of items in the shopping cart, we'll use the commercetools API to fetch and display items.

### CODE REVIEW

File: src/services/commercetoolsApi.ts
Comment (vetalapo) [inline src/services/commercetoolsApi.ts:1]: Please revert changes to this file.

File: src/views/BasketView.vue [hunk 1]
```diff
@@ -1,6 +1,90 @@
...
+export default defineComponent({
+    data() {
+        return {
+            authStore: useAuthStore(),
+            cartProducts: [] as LineItem[],
+            isLoading: false,
+            errorMessage: null as string | null
+        };
+    },
+    async mounted() {
```
Comment (vetalapo) [inline src/views/BasketView.vue:16]: Please remove everything in mounted()

File: src/views/BasketView.vue [hunk 2]
```diff
@@ -1,6 +1,90 @@
+<script lang="ts">
+import { defineComponent } from "vue";
+import { useAuthStore } from "@/store";
+import { getMyCarts } from "@/services/commercetoolsApi";
+import type { CartAPI, LineItem } from "@/types";
+
+export default defineComponent({
```
Comment (vetalapo) [inline src/views/BasketView.vue:7]: You don't have to define a component. Please make it consistent with existing views.

**Review by vetalapo — CHANGES_REQUESTED** [review]
Everything you need should be already in cart in local storage. There's no need to go fetch data at this point, only if you update something.

File: .env.example
Comment (vetalapo) [inline .env.example:1]: Why is this file removed? Cancel this change.

File: src/views/BasketView.vue [hunk 3]
```diff
@@ -1,6 +1,99 @@
...
+                cartProducts: [] as LineItem[],
+                isLoading: false,
+                errorMessage: null as string | null
+            };
+        },
+        methods: {
+            formatPrice(amount: number): string {
+                return (amount / 100).toFixed(2);
+            },
+            async fetchCartItems() {
```
Comment (vetalapo) [inline src/views/BasketView.vue:19]: There is no need to fetch data and store it in local storage as local storage already contains all data you might need for this task.

**Review by vetalapo — CHANGES_REQUESTED** [review]
Also please include a screenshot with cart items.

File: .env.example
Comment (vetalapo) [inline .env.example:1]: Discard changes to this file.

File: src/views/BasketView.vue [hunk 4]
```diff
@@ -1,6 +1,85 @@
+<script lang="ts">
+    import { useAuthStore } from "@/store";
+    import type { CartAPI, LineItem } from "@/types";
+
+    export default {
+        data() {
+            const storedCartDataString = localStorage.getItem("cartData");
```
Comment (vetalapo) [inline src/views/BasketView.vue:7]: There is no need to go to the local storage, AuthStore already has it ready to use.
Also please remove all this logic from data, it doesn't belong here.

**Review by vetalapo — CHANGES_REQUESTED** [review]
Please include an additional screenshot in PR description of the basket page with some items displayed. The task says it should be displayed, just want to make sure everything works.

File: .env.example
Comment (valeryiay) [reply .env.example:1]: the file was accidentally deleted and then restored

File: src/views/BasketView.vue (hunk 4 above)
Comment (valeryiay) [reply src/views/BasketView.vue]: Thank you!

**Comment by vetalapo** [comment]
Sorry, I'll have to close the PR. Making it work will be incredibly hard, considering time constraints we got.

**Pull request closed without merging**